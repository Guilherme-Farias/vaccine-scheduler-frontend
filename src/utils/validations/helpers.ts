import Joi from 'joi';
import { IParamErrors } from 'types/global';

export function getFieldErrors(objError: Joi.ValidationResult) {
  const errors: IParamErrors = {};

  if (objError.error) {
    objError.error.details.forEach(err => {
      errors[err.path.join('.')] = err.message;
    });
  }

  return errors;
}
