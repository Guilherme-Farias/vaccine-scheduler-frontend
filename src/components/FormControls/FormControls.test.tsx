import { render, screen } from 'utils/tests/helpers';

import { FormGroup, FormLabel, FormErrorMessage } from '.';

describe('<FormControls />', () => {
  it('should render the FormControls', () => {
    const { container } = render(
      <FormGroup>
        <FormLabel htmlFor="test">Test</FormLabel>
        <input id="test" />
        <FormErrorMessage />
      </FormGroup>,
    );
    expect(screen.getByRole('group')).toBeInTheDocument();
    expect(screen.getByLabelText(/test/i)).toBeInTheDocument();
    expect(screen.getByRole('textbox')).toBeInTheDocument();
    expect(screen.getByRole('alert')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
