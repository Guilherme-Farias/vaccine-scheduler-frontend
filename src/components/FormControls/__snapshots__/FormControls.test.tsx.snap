// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormControls /> should render the FormControls 1`] = `
.c0 {
  cursor: pointer;
  display: inline-block;
  color: #171923;
  font-size: 1rem;
  font-weight: 500;
  margin-bottom: 0.5rem;
}

.c1 {
  color: #DC2626;
  font-size: 0.875rem;
}

<div
  class=""
  role="group"
>
  <label
    class="c0"
    for="test"
  >
    Test
  </label>
  <input
    id="test"
  />
  <span
    class="c1"
    role="alert"
  />
</div>
`;
